hilbert/runners/run_base.py

```diff
@@ -122,9 +159,44 @@ def add_batch_size_arg(parser):
 def add_shard_factor_arg(parser):
     parser.add_argument(
         '--shard-factor', type=int, default=1,
-        help= "Divide sectors by shard_factor**2 to make it fit on GPU."
+        help="Divide sectors by shard_factor**2 to make it fit on GPU."
     )
 
+def add_remove_cooc_arg(parser):
+    parser.add_argument(
+        '--remove-threshold', '-thres', type=int, default=10, dest='remove_threshold',
+        help="A small number threshold of cooc counts to be removed to fit into the "
+             "GPU memory."
+    )
+
+def add_gradient_clipping_arg(parser):
+    parser.add_argument(
+        '--clipping', '-C', type=float, default=None, dest='gradient_clipping',
+        help="gradient clipping value."
+    )
+
+def add_LR_scheduler_arg(parser):
+    # can't have both LR and LR scheduler??
+    parser.add_argument(
+        '--LR-scheduler', '-scheduler', default=None, choices=['linear', 'inverse', 'None'], dest='scheduler_str',
+        help="Type of learning rate scheduler"
+    )
+    parser.add_argument(
+        '--LR-scheduler-endLR', '-le', type=float, default=0.0, dest='end_lr',
+        help="The end learning rate for linear learning rate scheduler"
+    )
+    parser.add_argument(
+        '--constant-fraction', '-frac', type=float, default=0.1, dest='constant_fraction',
```

Comment:
Name this to make it more linked to lr_scheduler.  e.g. `--lr_scheduler_fraction`.